torchao/dtypes/floatx/cutlass_semi_sparse_layout.py

```diff
@@ -210,3 +257,46 @@ def _linear_fp8_act_fp8_weight_sparse_cutlass_impl(input_tensor, weight_tensor,
     )
 
     return out
+
+
+def _linear_fp8_act_sparse_fp8_weight_cutlass_check(input_tensor, weight_tensor, bias):
+    from torchao.dtypes.floatx import Float8Layout
+
+    res = (
+        isinstance(input_tensor, AffineQuantizedTensor)
```

Comment:
just FYI, I feel we could simplify this list of checks if don't need to support a variety of kernels that all use the same subclassed tensor, it only needs to be specific enough so we can dispatch to the correct kernel. maybe this can be simplified if we split the AQT in the future.